xarray/coding/calendar_ops.py

```diff
@@ -0,0 +1,280 @@
+        raise ValueError(
+            f"Source time coordinate contains dates with year 0, which is not supported by target calendar {tgt_cal}."
+        )
+
+    if (source == "360_day" or calendar == "360_day") and align_on is None:
+        raise ValueError(
+            "Argument `align_on` must be specified with either 'date' or "
+            "'year' when converting to or from a '360_day' calendar."
+        )
+
+    if source != "360_day" and calendar != "360_day":
+        align_on = "date"
+
+    out = ds.copy()
+
+    if align_on == "year":
+        # Special case for conversion involving 360_day calendar
+        # Instead of translating dates directly, this tries to keep the position within a year similar.
+        def _yearly_interp_doy(time):
+            # Returns the nearest day in the target calendar of the corresponding "decimal year" in the source calendar
+            yr = int(time.dt.year[0])
+            return np.round(
+                _days_in_year(yr, calendar, use_cftime)
+                * time.dt.dayofyear
+                / _days_in_year(yr, source, use_cftime)
+            ).astype(int)
+
+        def _convert_datetime(date, new_doy, calendar):
+            """Convert a datetime object to another calendar.
+
+            Redefining the day of year (thus ignoring month and day information from the source datetime).
+            Nanosecond information are lost as cftime.datetime doesn't support them.
+            """
+            new_date = cftime.num2date(
+                new_doy - 1,
+                f"days since {date.year}-01-01",
+                calendar=calendar if use_cftime else "standard",
+            )
+            try:
+                return get_date_type(calendar, use_cftime)(
+                    date.year,
+                    new_date.month,
+                    new_date.day,
+                    date.hour,
+                    date.minute,
+                    date.second,
+                    date.microsecond,
+                )
+            except ValueError:
+                return np.nan
+
+        new_doy = time.groupby(f"{dim}.year").map(_yearly_interp_doy)
+
+        # Convert the source datetimes, but override the doy with our new doys
+        out[dim] = DataArray(
+            [
+                _convert_datetime(date, newdoy, calendar)
+                for date, newdoy in zip(time.variable._data.array, new_doy)
+            ],
+            dims=(dim,),
+            name=dim,
+        )
+        # Remove duplicate timestamps, happens when reducing the number of days
+        out = out.isel({dim: np.unique(out[dim], return_index=True)[1]})
+    elif align_on == "date":
+        new_times = convert_times(
+            time.data,
+            get_date_type(calendar, use_cftime=use_cftime),
+            raise_on_invalid=False,
+        )
+        out[dim] = new_times
+
+        # Remove NaN that where put on invalid dates in target calendar
+        out = out.dropna(dim)
```

Comment:
Sorry I think I was wrong about this before -- I think this may only look for NaN values in the data, not the coordinate, so let's restore what was here before:
```suggestion
        out = out.where(out[dim].notnull(), drop=True)
```

Reply:
Oh you are right!